deep_qa/run.py

```diff
@@ -0,0 +1,160 @@
+        logging.getLogger().addHandler(handler)
+        shutil.copyfile(param_path, log_dir + "_model_params.json")
+
+    if model_class is None:
+        model_type = params.pop_choice('model_class', concrete_models.keys())
+        model_class = concrete_models[model_type]
+    else:
+        if params.pop('model_class', None) is not None:
+            raise ConfigurationError("You have specified a local model class and passed a model_class argument"
+                                     "in the json specification. These options are mutually exclusive.")
+    model = model_class(params)
+
+    if model.can_train():
+        logger.info("Training model")
+        model.train()
+        K.clear_session()
+    else:
+        raise ConfigurationError("The supplied model does not have enough training inputs.")
+
+
+def load_model(param_path: str, model_class=None):
+    """
+    Loads and returns a model.
+
+    Parameters
+    ----------
+    param_path: str, required
+        A json file specifying a DeepQaModel.
+    model_class: DeepQaModel, optional (default=None)
+        This option is useful if you have implemented a new model
+        class which is not one of the ones implemented in this library.
+
+    Returns
+    -------
+    A ``DeepQaModel`` instance.
+    """
+    param_dict = pyhocon.ConfigFactory.parse_file(param_path)
+    params = Params(replace_none(param_dict))
+    prepare_environment(params)
+
+    from deep_qa.models import concrete_models
+    if model_class is None:
+        model_type = params.pop_choice('model_class', concrete_models.keys())
+        model_class = concrete_models[model_type]
+    else:
+        if params.pop('model_class', None) is not None:
+            raise ConfigurationError("You have specified a local model class and passed a model_class argument"
+                                     "in the json specification. These options are mutually exclusive.")
+    model = model_class(params)
+    model.load_model()
+    return model
+
+
+def evaluate_model(param_path: str, test: str=False, model_class=None):
+    """
+    Loads and evaluates a dataset from a saved parameter path.
+
+    Parameters
+    ----------
+    param_path: str, required
+        A json file specifying a DeepQaModel.
+    test: str, optional, (default=False)
+        Whether to use the test data or validation data to evaluate the model.
+    model_class: ``DeepQaModel``, optional (default=None)
+        This option is useful if you have implemented a new model class which
+        is not one of the ones implemented in this library.
+
+    Returns
+    -------
+    Numpy arrays of model predictions in the format of model.outputs.
+
+    """
+    model = load_model(param_path, model_class=model_class)
+    dataset_files = model.test_files if test else model.validation_files
+    dataset = model.load_dataset_from_files(dataset_files)
+    indexed_dataset = dataset.to_indexed_dataset(model.data_indexer)
```

Comment:
Ah, slight issue here: `Trainer` doesn't define a `data_indexer`, `TextTrainer` does, so we can't index the dataset in `Trainer`. Are you happy with this going in `TextTrainer` instead? It feels a bit weird given we do the training etc  in `Trainer`...